### ISSUE TITLE
Binary in Ubuntu PPA is over a year old, and is targeted at Ubuntu versions that have reached end-of-life

### ISSUE THREAD

**Issue by patrick-brian-mooney** [pithos/pithos#340]
Pithos 1.2.1 is out, but the version delivered in the Ubuntu PPA is 1.0.3.

Surely new builds could be uploaded to the PPA? The description of 1.0.3 on the home page says it is "for those of you using Fedora <= 20 and Ubuntu <= 14.10." Ubuntu 14.10 reached end of life over a year ago. Surely supporting these non-long-term-support people is not such a high priority as to hold back all users of the Pithos Ubuntu PPA to a version of Pithos that was superseded by another point release over a year ago.

**Comment by TingPing** [comment]
This was mentioned in #337 but 14.04 is too old for the current release.

**Comment by patrick-brian-mooney** [comment]
This is not a duplicate of #337: it's the opposite problem of the one in that bug. #337 complains that distributing 1.0.3 inconveniences people who can't upgrade to Ubuntu 14.04. My complaint is that targeting 14.04 is targeting a version that's obsolete, because 14.04 is already past its end of life. Or, to put it another way, #337 complains that the version in the Ubuntu PPA is too new; my complaint is that it's waaaay too old.

**Comment by TingPing** [comment]
> 14.04 is already past its end of life

It is an LTS release "supported" until 2019: http://www.ubuntu.com/info/release-end-of-life

**Comment by patrick-brian-mooney** [comment]
Whoops, so it is. That's what I get for working from secondary sources. =\

Nevertheless, the version offered in the PPA is over a year old (it's impossible to tell for sure, because 1.0.3 doesn't even appear in the changelog). Making a later version available would make a lot of bug fixes available by default without requiring people who want a newer version to build from scratch.

I would like to humbly suggest that if there's a point to providing a repo at all, then that repo should provide more or less up-to-date software; and if it's not possible or not practical to provide multiple versions in the repo, then maybe building from scratch should be done by the people who want to install older versions, rather than people who want bug fixes.

**Comment by TingPing** [comment]
> it's impossible to tell for sure, because 1.0.3 doesn't even appear in the changelog

Except it is...

> Making a later version available would make a lot of bug fixes available by default without requiring people who want a newer version to build from scratch.

It can't be built on 14.04 as I already said it is too old.

> I would like to humbly suggest that if there's a point to providing a repo at all, then that repo should provide more or less up-to-date software

The official repos have version 0.3.17 which is actually *4* years old and to my knowledge can't actually connect to Pandora, at least not Pandora-One users. So while 1.0.3 is one year old it is a much better version.

**Comment by patrick-brian-mooney** [comment]
Yes, but 1.2.1 is more up to date, and surely is a better choice for most users. Otherwise, why offer it in the first place?

> It [1.0.3] can't be built on 14.04 as I already said it is too old.

My suggestion is that 1.2.1 be the version offered in the repo. The whole point of offering something in a repo is that it doesn't need to be built; the version users get from the repo *is already built*. Generally, this should be the most up-to-date version, no?

What is the actual downside to putting 1.2.1 in the repo, and having people who genuinely need 1.0.3 build from source?

**Comment by TingPing** [comment]
> Yes, but 1.2.1 is more up to date, and surely is a better choice for most users. Otherwise, why offer it in the first place?

For the last time: It does not build on 14.04 because it is too old.

**Comment by patrick-brian-mooney** [comment]
Right. Got that. 

What does that have to do with the pre-built version offered in the repos?

**Comment by TingPing** [comment]
There are two options:

- Don't have a PPA, which leaves users with a very outdated version
- Have a PPA, which leaves users with a less outdated version

There is no third option of giving them a newer version because it cannot be ran on such old software.

**Comment by patrick-brian-mooney** [comment]
That's a false dichotomy: 

> Have a PPA, which leaves users with a less outdated version

The PPA *could, in fact*, offer an up-to-date version, rather than "a less outdated version." Additionally, [it's possible](http://askubuntu.com/questions/30145/ppa-packaging-having-versions-of-packages-for-multiple-distros) for a PPA to target multiple distros or multiple versions of a distro, so "we're stuck with it only being possible to target one version of Ubuntu, because it's actually impossible to do anything else, so we've decided to target 14.04" is based on a flawed premise. I mean, if you want to offer "a less outdated version," and are willing to set up a PPA to do so, does it not make sense to offer the *least* outdated version possible?

> There is no third option of giving them a newer version because it cannot be ran on such old software.

That's not true, either: Currently, multiple versions are offered; it's just that all but one version requires compiling from source. My suggestion is that the decision about which version should be available to people who want to avoid compiling from source should be tied more closely to *which version is most bug-free* rather than *which version is available for the earliest-supported operating system*. 

The current version distribution strategy is:

* version 1.0.3 is available in the PPA.
* versions 1.1.1, 1.1.2, 1.2.0, 1.2.1, and all pre-1.0.3 versions are available by compiling from source.

This graces 1.0.3 as a default version for many users. I would like to politely suggest that every bug report that comes in that results in a "can you try compiling the current version from source and see if the error still occurs" message is an effect of this strategy: providing a PPA in the first place is a service precisely insofar as it offers more up-to-date software.

Here is what I have been trying to suggest:

* The PPA contains version 1.2.1, and if a new stable version is produced later, then the PPA is updated to contain that version.
* All other versions are *still available to everyone*, including people who don't want to upgrade to a newer version of their OS. They just require compiling from source.

I mean, Jesus, it's your software. You're providing it to the world for free, and you don't owe anyone anything. If you want to not provide a newer version in the PPA, don't. You can absolutely say "we don't have anyone who's willing to compile that version for the right architectures, and that's why we're sticking with 1.0.3." Or you can say "We're not going to do it because we don't feel like it," and that's your prerogative, too. You can even say "We're not going to do it because fuck you, that's why," and that's also fair: like I said, it's your software, and you can package and distribute it in any way that you want. It's your prerogative. I was just making a friendly suggestion in the hope that it would be helpful.

But do me the favor of not bullshitting by saying that what I suggested is impossible, because that's not true.

**Comment by TingPing** [comment]
Just so we are on the same page, you do realize for 16.04 the PPA does package 1.2.1 correct?

https://launchpad.net/~pithos/+archive/ubuntu/ppa/+build/10543362

**Comment by patrick-brian-mooney** [comment]
No, I didn't! I'm running 16.04 (well ... actually, I'm running Linux Mint Sarah, based on Ubuntu 16.04), and when I reinstalled the OS and followed the Pithos install directions on the front page back in early July, I got 1.0.3 installed. Looking at `/etc/apt/sources.list.d/pithos*` shows me the only file is `pithos-ppa-trusty.list`, and it contains

    deb http://ppa.launchpad.net/pithos/ppa/ubuntu trusty main
    deb-src http://ppa.launchpad.net/pithos/ppa/ubuntu trusty main

However, removing the PPA and re-adding it got me the expected version for Xenial, so I suppose it's a moot point now; maybe I managed to re-install before the Xenial repo became available? Or maybe my idiotic ISP had cache problems?

In any case, the problem seems to be fixed now. Thank you for the help!
